docs/dev.rst

```diff
@@ -255,26 +255,30 @@ You should then run the development server as usual, with ``start``. Every chang
 Running edx-platform unit tests
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-It's possible to run the full set of unit tests that ship with `edx-platform <https://github.com/openedx/edx-platform/>`__. To do so, run a shell in the LMS development container::
+It's possible to run the full set of unit tests that ship with `edx-platform <https://github.com/openedx/edx-platform/>`__. Some tests are intended to be run in an LMS context, some are intended for the CMS context, and some can be run in both. To run tests for the LMS, start a shell in the LMS container::
 
     tutor dev run lms bash
 
-Then, run unit tests with ``pytest`` commands::
+Then, simply `invoke pytest <https://docs.pytest.org/en/latest/how-to/usage.html>`_. For example::
 
-    # Run tests on common apps
-    unset DJANGO_SETTINGS_MODULE
-    unset SERVICE_VARIANT
-    export EDXAPP_TEST_MONGO_HOST=mongodb
-    pytest common
-    pytest openedx
+    pytest lms                                              # Run tests for the entire LMS source tree.
+    pytest lms/djangoapps/ccx                               # Run tests in a single LMS app.
+    pytest lms/djangoapps/ccx -k '_override or _structure'  # Run tests in a single LMS app with names matching substring(s).
+    pytest openedx common                                   # Run tests for both shared source trees.
+    pytest openedx/core/djangoapps/content/course_overviews # Run tests for a single shared app.
+    # ... etc.
 
-    # Run tests on LMS
-    export DJANGO_SETTINGS_MODULE=lms.envs.tutor.test
-    pytest lms
+To run CMS tests, start a shell in the CMS container::
 
-    # Run tests on CMS
-    export DJANGO_SETTINGS_MODULE=cms.envs.tutor.test
-    pytest cms
+    tutor dev run cms bash
+
+As before, simply invoke pytest. Tests under the cms/ source tree will automatically use CMS settings. For tests under the shared source trees, you will need to tell pytest to look in the cms/ directory for configuration by providing ``--rootdir`` option::
+
+    pytest cms                                                            # Run tests for the entire CMS source tree.
+    pytest cms/djangoapps/coursegraph                                     # Run tests in a single CMS app.
+    pytest --rootdir=cms openedx common                                   # Run tests for both shared source trees.
+    pytest --rootdir=cms openedx/core/djangoapps/content/course_overviews # Run tests for a single shared app.
```

Comment:
This contradicts the instruction from line 268. I'm confused: do we need to specify the rootdir when running shared app tests?

Reply:
@regisb Right, sorry, it's a little convoluted.

By default, ALL tests will use `lms.envs.test` (because of edx-platform's setup.cfg) EXCEPT the tests under `cms/`, which will use `cms.envs.test` (because of edx-platform's cms/pytest.ini).

This is generally reasonable, except that sometimes folks will want to run tests for the "shared" directories (openedx/ and common/) in a CMS context. The easiest way to make this happen is to tell pytest to look at the cms/ directory for a pytest.ini, hence `--rootdir=cms`.

As I think more about this, I really wish we could just have the behavior of:
* tests in lms/ always use `lms.envs.test`,
* tests in cms/ always use `cms.envs.test`,
* tests in shared directories use `lms.envs.test` when in the LMS container, and use `cms.envs.test` when in the CMS container.

Orchestrating that will take some hair-pulling. For the conference, I think we'll just have to accept the status quo. The current instructions, while a little annoying, certainly work for the vast majority of edx-platform tests :)